server/publications/collections/orders.js

```diff
@@ -63,10 +65,51 @@ Meteor.publish("CustomPaginatedOrders", function (query, options) {
   if (!shopId) {
     return this.ready();
   }
+
+  // return any order for which the shopId is attached to an item
+  const selector = {
+    "items.shopId": shopId
+  };
   if (Roles.userIsInRole(this.userId, ["admin", "owner", "orders"], shopId)) {
-    Counts.publish(this, "order-count", Orders.find({ shopId: shopId }), { noReady: true });
-    return Orders.find({ shopId: shopId });
+    ReactiveAggregate(this, Orders, [
+      {
+        $project: {
+          items: {
+            $filter: {
+              input: "$items",
+              as: "item",
+              cond: { $eq: ["$$item.shopId", shopId] }
```

Comment:
Let's consider using `$in` and passing in an array of shopIds instead of checking against one shopId. That would permit marketplace owners or multi-shop admins to see several shops rather than just one.

That being said, it might make the UI far more complicated, so it's possible that it's better to permit those types of users to switch between the shops they are viewing the order as.

@rymorgan any thoughts on this?

Reply:
In Mongo3.2, `$in` is not supported in projections. It is in 3.4